crypto/finite_field.ts

```diff
@@ -199,12 +199,17 @@ function createField(
     fromBigint(x: bigint) {
       return mod(x, p);
     },
-    rot(x: bigint, bits: number, direction: 'left' | 'right' = 'left') {
+    rot(
+      x: bigint,
+      bits: number,
+      direction: 'left' | 'right' = 'left',
+      maxBits = 64
+    ) {
```

Comment:
maybe we can use native bit shifts to make the code faster and cleaner?

```ts
    rot(
      x: bigint,
      bits: bigint,
      direction: 'left' | 'right' = 'left',
      maxBits = 64n
    ) {
      if (direction === 'right') bits = maxBits - bits;
      let full = x << bits;
      let excess = full >> maxBits;
      let shifted = full & ((1n << maxBits) - 1n);
      return shifted | excess;
    } 
```

Reply:
yep, good idea!